src/commons/utils/cluster_utils.py

```diff
@@ -416,3 +417,54 @@ def get_all_workers_details(self, names: bool = True) -> list:
             resp = resp[1].strip().split("\n")[1:]
             LOGGER.info("worker nodes: %s", resp)
         return resp
+
+    def get_cluster_config(self, pod_list=None) -> dict:
+        """
+        Function to fetch data from file (e.g. conf files).
+
+        :param pod_list: Data pod name list to get the cluster.conf File.
+        """
+        if pod_list is None:
+            pod_list = self.get_all_pods(pod_prefix=const.DATA_POD_NAME_PREFIX)
+        pod_name = random.sample(pod_list, 1)[0]
+        conf_cp = cmd.K8S_CP_TO_LOCAL_CMD.format(pod_name, const.CLUSTER_CONF_PATH,
+                                                 const.LOCAL_CONF_PATH, const.HAX_CONTAINER_NAME)
+        resp_node = self.execute_command(command=conf_cp)
+        LOGGER.debug("%s response %s ", conf_cp, resp_node)
+        local_conf = os.path.join(os.getcwd(), "cluster.conf")
+        if os.path.exists(local_conf):
+            os.remove(local_conf)
+        self.download_file(local_path=local_conf, remote_path=const.LOCAL_CONF_PATH)
+
+        return get_config_yaml(local_conf)
+
+    def retrieve_durability_values(self, dur_type: str = 'sns') -> dict:
+        """
+        Return the durability Configuration for Data/Metadata (SNS/DIX) for the cluster.
+
+        :param dur_type: sns/dix
+        :return : dict of format { 'data': '1','parity': '4','spare': '0'}
+        """
+        resp = self.get_cluster_config()['cluster']['storage_set'][0]['durability'][
+            dur_type.lower()]
+        LOGGER.info(resp)
+        return resp
+
+    def get_user_quota_in_bytes(self):
+        """Get available user quota in bytes"""
+        status, cluster_stat = self.check_cluster_storage()
+        if not status:
+            raise AssertionError(f"Failed to get cluster storage details: {cluster_stat}")
+        total_capacity = cluster_stat.get("total_capacity")
+        avail_capacity = cluster_stat.get("avail_capacity")
+        used_capacity = cluster_stat.get("used_capacity")
+        LOGGER.info("Total Capacity: %s Available Capacity: %s Used Capacity: %s", total_capacity,
+                    avail_capacity, used_capacity)
+        # Get SNS configuration to retrieve available user data space
+        durability_values = self.retrieve_durability_values('sns')
+        sns_values = {key: int(value) for key, value in durability_values.items()}
+        LOGGER.debug("Durability Values (SNS) %s", sns_values)
+        user_quota_to_writes = int(sns_values['data'] / sum(sns_values.values()) * avail_capacity)
```

Comment:
The user quota to write should be calculated based on total capacity and not available capacity.

Reply:
Hi @priyankaborawake 
I am not convinced about total capacity for cluster.
**Reason:** 
1. We can not consider already filled capacity (written by some other user), hence our total capacity will be anyway available capacity.
2. Our read percentage will be considered on the basis of written data by us, we are not having control on already written data. We will read whatever we write, not the data which is already written. 
3. i.e. If total storage capacity is 100 GB and 50% data is written/filled, hence available space will be 50GB, So lets say Write % is 90%, Read % is 100% then we need to consider 50 GB, as Total space hence we will use 45 GB for write and same 45 GB for read which is again calculated on the basis of written data not on the 100GB.
4. Also Ref Type 3 scenario, Ref Below scenario where **user data capacity** means available capacity.
![image](https://user-images.githubusercontent.com/66417432/181765668-db429b24-3f9f-4d81-b922-20d2d745dfc3.png)
5. Also full cluster capacity  may not be available for user in future and user need to work on given capacity only.
As of now we are merging this PR for IO executions. and can discuss for any additional requirements as needed.